pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -470,7 +470,7 @@ func (dr *DashboardServiceImpl) setDefaultPermissions(ctx context.Context, dto *
 	inFolder := dash.FolderId > 0
 	if !accesscontrol.IsDisabled(dr.cfg) {
 		var permissions []accesscontrol.SetResourcePermissionCommand
-		if !provisioned {
+		if !provisioned && dto.User.IsRealUser() && !dto.User.IsAnonymous {
```

Comment:
We could only check for `IsRealUser()` as anonymous users have an ID set to `0` in the context which we consider as non real users. But in case some day we endup changing their ID, I added both checks, just in case.

Reply:
I think that's a good shout! 👌